pkg/kv/kvserver/kvflowcontrol/replica_rac2/processor.go

```diff
@@ -892,42 +856,38 @@ func (p *processorImpl) AdmitRaftEntriesRaftMuLocked(ctx context.Context, e rac2
 func (p *processorImpl) EnqueuePiggybackedAdmittedAtLeader(
 	from roachpb.ReplicaID, state kvflowcontrolpb.AdmittedState,
 ) {
-	p.mu.Lock()
-	defer p.mu.Unlock()
-	if p.mu.leader.rc == nil {
+	if p.leader.rc == nil {
```

Comment:
@sumeerbhola This should be read with `rcReferenceUpdateMu`. We have a few options:

1. Lock it and check `rc == nil` check below `pendingAdmittedMu.Lock`. Downside is that we're blocking `rcReferenceUpdateMu` unnecessarily while updating `pendingAdmitted` map.
2. Lock it narrowly to check `rc != nil`. Though if we unlock it, `rc` can become nil and nil the map. So we'll need to check `pendingAdmitted != nil` below.
3. So, it's better to have an invariant `pendingAdmitted != nil ==> rc != nil`. Then we don't need to check `rc` here, and will just check `pendingAdmitted != nil` below.

Reply:
Went with (3), and made is so that we don't ever need to hold both `rcReferenceUpdateMu` and `pendingAdmittedMu` simultaneously.